specs/withdrawal-proof.md

```diff
@@ -0,0 +1,74 @@
+
+The withdrawals proof verifies the merkle patricia trie identified by the root `withdrawalsRoot` contains all the withdrawals.
+
+`Withdrawal` operation is introduced by EIP-4895 to provides a way for validator withdrawals made on the beacon chain to enter into the EVM. `Withdrawal` operation is a system-level operation and introduces a new field after the existing fields in the execution payload. Unlike transactions, there is no signature to verify and the main verification logic is to verify withdrawals MPT root.
+
+## Withdrawal encoding
+
+The execution payload gains a new field for the withdrawals which is an RLP list of Withdrawal data. This new field is encoded after the existing fields in the execution payload structure and is considered part of the execution payload’s body.
+
+```
+withdrawal_0 = [withdrawal_index_0, validator_index_0, address_0, amount_0]
+withdrawal_1 = [withdrawal_index_1, validator_index_1, address_1, amount_1]
+withdrawals = [withdrawal_0, withdrawal_1]
+
+execution_payload_body_rlp = RLP([transactions, [], withdrawals])
+```
+
+
+## Circuit behavior
+
+For every withdrawal defined as the parameters `(withdrawal_index, validator_index, address, amount)` and the circuit verifies the followings:
+
+1. `withdrawalsData: bytes = rlp([withdrawal_index, validator_index, address, amount])`
+2. `withdrawalsRoot: word = mpt(withdrawalsData)`
+3. `withdrawal_index`, `validator_index` and `amount` are all `uint64` values.
+4. `amount_wei = amount * 1e9` and increases validator's balance by `amount_wei`
+
+- The rlp encoding of withdrawal parameters will be done using a custom rlp encoding gadget,  isolated from the rlp encoding used by the MPT circuit.
+- The MPT root verification will be done in MPT circuit; the withdrawal circuit do a lookup to the MPT table.
+
+From this information the circuit builds the WDTable:
+
+Where:
+
+- Address = validator's address
+- Amount = a nonzero amount of ether given in Gwei (1e9 wei)
+
+| 0 WithdrawalID | 1 Validator ID | 2 Tag               | 3 value       |
+| -----------    | -------------  | ----------------    | ------------- |             
+|                |                | *WDContextFieldTag* |               |
+| $WithdrawalID  | $ValidatorID   | Address             | $value{Lo,Hi} |
+| $WithdrawalID  | $ValidatorID   | Amount              | $value{Lo,Hi} |
+
+There are some constraints on the shape of the table like:
+
+- For every withdrawal, each tag must appear exactly once.
+- `WithdrawalID` is increased monotonically and sequentially for each withdrawal. Which means it won't be starting from 0 in most of cases.
+
+Since the withdrawal table is built from public inputs, the verifier (the L1 smart contract in the zkRollup for example) needs to validate that all rows of the table are properly built using the withdrawal data.  Since the table construction is validated outside of the circuit, there's no need to verify the same constraints inside of the circuit.
```

Comment:
`WithdrawalRoot` contains ONLY the withdraws operations of this block. So I think public input we just need to add one extra field `WithdrawalRoot`, and the withdrawal table can just be witness from prover, and rely on the validation of `mpt(withdrawalsData)` via lookup into mpt table.

Reply:
Based on this, in `super_circuit.png` public input for withdrawls can just one single 256 bit field `WithdrawalRoot` :)

Reply:
`super_circuit.png` updated in [e19a56c](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/487/commits/e19a56cf56c3060ea4e0be58d080daa1d17c5d22)